packages/react-native/src/components/3DS/useThreeDSecure.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from "react";
+import { useRef } from "react";
+import { useEvervault } from "../EvervaultProvider";
+import { startSession, threeDSecureSession } from "./threeDSSession";
+import {
+  ThreeDSecureCallbacks,
+  UseThreeDSecureResponse,
+  UseThreeDSecureState,
+} from "./types";
+
+export const useThreeDSecure = (): UseThreeDSecureResponse => {
+  const { appUuid } = useEvervault();
+  const intervalRef = useRef<NodeJS.Timeout | null>(null);
+  const [displayModal, setDisplayModal] = useState(false);
+  const [state, setState] = useState<UseThreeDSecureState>({
+    session: null,
+    displayModal,
+  });
```

Comment:
No need to duplicate the modal state here. Instead we can just have.

```
const [session, setSession] = useState(null)
const [isVisible, setIsVisible] = useState(flase);
```

And then call the setSession and setIsVisible methods individually.

The return from the hook would become

```
return {
    start,
    cancel,
    session,
    isVisible
};
```